diff-hl.el

```diff
@@ -683,6 +687,51 @@ The value of this variable is a mode line template as in
               (t (memq major-mode diff-hl-global-modes)))
     (turn-on-diff-hl-mode)))
 
+;;;###autoload
+(defun diff-hl-set-reference-rev (&optional rev)
+  "Set the reference revision globally to REV.
+When called interactively, REV is get from contexts:
+
+- In a log view buffer, it uses the revision of current entry.
+Call `vc-print-log' or `vc-print-root-log' first to open a log
+view buffer.
+- In a VC annotate buffer, it uses the revision of current line.
+- In other situations, get the revision name at point.
+
+Notice that this sets the reference revision globally, so in
```

Comment:
Maybe if this turns out to be a real problem, the "reference revision" structure could evolve into an alist, mapping repository roots to reference revisions.